Reactive-Network/reactive-bounties
# Implemented bounty "Automated Funds Distribution — 400"

# Pull Request: Implement Multi-Party Wallet with Reactive Smart Contract

## Description
This pull request implements a Multi-Party Wallet use case using Reactive Smart Contracts. It includes:

- `MultiPartyWallet.sol`: The main contract on Sepolia
- `MemeCoin.sol`: An ERC20 token contract for the meme coin
- `MultiPartyWalletReactive.sol`: The Reactive Smart Contract
- Deploy scripts for each contract
- Instructions for deployment and interaction

## Changes Made
- Added `MultiPartyWallet.sol` contract
- Added `MemeCoin.sol` contract
- Added `MultiPartyWalletReactive.sol` contract
- Added deploy scripts for each contract
- Added `workflow.md` detailing the step-by-step execution process
- Added `Cover_Letter.md` explaining what is being done `src/Cover_Letter.md` 
- Updated README.md with setup and usage instructions

## How to Test
1. Deploy the contracts as described in the workflow
2. Follow the steps in the workflow to interact with the contracts
3. Verify that all events are emitted and reactive actions are triggered as expected

## Additional Notes
- All contracts have been tested on Sepolia and Reactive testnets
- Transaction hashes for each step are provided in the workflow document
